apis/clusterresources/v1beta1/clusterbackup_types.go

```diff
@@ -27,7 +27,6 @@ import (
 
 // ClusterBackupSpec defines the desired state of ClusterBackup
 type ClusterBackupSpec struct {
-	ClusterID   string `json:"clusterId"`
 	ClusterKind string `json:"clusterKind"`
 }
```

Comment:
`ClusterKind` you can move simply into the `Status` and thus you will remove and simplify the code that calculates the appropriate Kind. 
Another consideration is whether triggering a backup through a cluster spec enhances the user experience. Need to ask what others think.